L1Trigger/Phase2DAQProducer/plugins/Phase2DAQProducer.cc

```diff
@@ -0,0 +1,218 @@
+
+        if (detUnit)
+        {
+            
+            bool isPSModulePixel = trackerGeometry.getDetectorType(detId) == TrackerGeometry::ModuleType::Ph2PSP;
+            bool isPSModuleStrip = trackerGeometry.getDetectorType(detId) == TrackerGeometry::ModuleType::Ph2PSS;
+            bool is2SModule      = trackerGeometry.getDetectorType(detId) == TrackerGeometry::ModuleType::Ph2SS;
+
+            if (isPSModuleStrip)
+            {
+                processClusters(TrackerGeometry::ModuleType::Ph2PSS, DetectorClusterCollection, dtc_id, slink_id, slink_id_within, dtcAssembly);
+            }
+            else if (isPSModulePixel)
+            {
+                processClusters(TrackerGeometry::ModuleType::Ph2PSP, DetectorClusterCollection, dtc_id, slink_id, slink_id_within, dtcAssembly);
+            }
+            else if (is2SModule)
+            {
+                processClusters(TrackerGeometry::ModuleType::Ph2SS, DetectorClusterCollection, dtc_id, slink_id, slink_id_within, dtcAssembly);
+            }
+
+        }
+    }
+
+    int i = 1;
+
+    // for (int i = 1; i == 1; ++i)
+    // {
+
+        DTCUnit& DTC_0 = dtcAssembly.GetDTCUnit(i); // dtc unit
+        
+        std::vector<std::vector<Cluster>> SLink0 = DTC_0.getClustersOnSLink(0);
+        std::vector<std::vector<Cluster>> SLink1 = DTC_0.getClustersOnSLink(1);
+        std::vector<std::vector<Cluster>> SLink2 = DTC_0.getClustersOnSLink(2);
+        std::vector<std::vector<Cluster>> SLink3 = DTC_0.getClustersOnSLink(3);
+        
+        DTC_0.convertToRawData(0);
+        DTC_0.convertToRawData(1);
+        DTC_0.convertToRawData(2);
+        DTC_0.convertToRawData(3);
+
+        // std::cout << (int)DTC_0.getDTCType() << std::endl;
+        
+        fedRawDataCollection.get()->FEDData( 0 + 4 * (i - 1) + 0 ) = DTC_0.GetSLink(0);
+        fedRawDataCollection.get()->FEDData( 1 + 4 * (i - 1) + 0 ) = DTC_0.GetSLink(1);
+        fedRawDataCollection.get()->FEDData( 2 + 4 * (i - 1) + 0 ) = DTC_0.GetSLink(2);
+        fedRawDataCollection.get()->FEDData( 3 + 4 * (i - 1) + 0 ) = DTC_0.GetSLink(3);
+
+    // }
+
+    iEvent.put(std::move(fedRawDataCollection));
+
+}
+
+void Phase2DAQProducer::processClusters(TrackerGeometry::ModuleType moduleType,
+                                           const Phase2TrackerCluster1DCollectionNew::DetSet& DetectorClusterCollection,
+                                           unsigned int dtc_id, unsigned int slink_id, unsigned int slink_id_within,
+                                           DTCAssembly& dtcAssembly)
+{
+    for (const auto& cluster : DetectorClusterCollection)
+    {
+        DTCUnit& assignedDtcUnit = dtcAssembly.GetDTCUnit(dtc_id);
+
+        unsigned int z = cluster.column();
+        unsigned int x = cluster.center();
+        unsigned int width = cluster.size();
+        unsigned int chipId = x % 127;
+        unsigned int sclusterAddress = 0;
+        unsigned int mipbit = 0;
```

Comment:
Surely mipbit = cluster.threshold()?

Reply:
have to check this, put it as a placeholder.